File: docs/user/en/query/requirements/examples/hierarchy-parents.java
final EvitaResponse<SealedEntity> result = session.querySealedEntity(
	query(
		// query hierarchy entity type
		collection("Product"),
		// target "True wireless" category
		filterBy(
			hierarchyWithin(
				"categories",
				attributeEquals("code", "true-wireless")
			)
		),
		require(
			hierarchyOfSelf(
				// request computation of all the parents of the "True wireless" category
				parents(
					"parentAxis",
					entityFetch(attributeContent(code)),
					statistics(
						CHILDREN_COUNT,
						QUERIED_ENTITY_COUNT
					)
				)
			)
		)
	)
);

final Hierarchy hierarchyResult = result.getExtraResult(Hierarchy.class);
// mega menu listing
final List<LevelInfo> megaMenu = hierarchyResult.getReferenceHierarchy("categories", "parentAxis");

File: documentation/user/en/query/requirements/examples/hierarchy/hierarchy-parents.cs
EvitaResponse<SealedEntity> entities = evita.QueryCatalog(
	"evita",
	session => session.QuerySealedEntity(
        Query(
        	Collection("Product"),
        	FilterBy(
        		HierarchyWithin(
        			"categories",
        			AttributeEquals("code", "true-wireless")
        		)
        	),
        	Require(
        		HierarchyOfReference(
        			"categories",
        			RemoveEmpty,
        			Parents(
        				"parentAxis",
        				EntityFetch(
        					AttributeContent("code")
        				),
        				Statistics(WithoutUserFilter, ChildrenCount, QueriedEntityCount)
        			)
        		)
        	)
        )
	)
);